project/sbt-changeling/Changeling.scala

```diff
@@ -0,0 +1,280 @@
+package systems.informal.sbt.changeling
+
+/**
+ * =Changeling=
+ *
+ * Changeling is an SBT pluging to make allow simple maintenance of a merge-conflict free changelog.
+ *
+ * ==Motivation==
+ *
+ * Merge conflicts arise when the same lines in the same files are edited
+ *
+ * "simultaneously" in two different branches. The basic idea is to reduce the chance of merge conflicts by adding each
+ * entry of an unreleased change into the changelog via a separate file.
+ *
+ * ===Design===
+ *
+ * In order to produce a changelog that has the usual and expected format in a flatfile, we just need a representation
+ * in the file system that will let us construct the expected result. If we look at a changelog, we can see an evident
+ * document model. Our changelog has roughly the following general shape:
+ *
+ * {{{
+ * # Changelog
+ *
+ * ## Unrelased
+ *
+ * ### Features
+ *
+ * - Change entry 1
+ * - Change entry 2
+ *
+ * ### Bug fixes
+ *
+ * - Change entry 3
+ * - Change entry 4
+ *
+ * ## 0.0.1
+ *
+ * ### Features
+ *
+ * - Change entry 1
+ * - Change entry 2
+ *
+ * ### Bug fixes
+ *
+ * - Change entry 3
+ * - Change entry 4
+ * }}}
+ *
+ * The obvious, high-level document model is a tree of depth 4:
+ *
+ * {{{
+ * Changelog
+ * |------- Unreleased
+ * |          |--------- Features
+ * |          |            |------ Change entry 1
+ * |          |            \------ Change entry 2
+ * |          |
+ * |          \--------- Bug fixes
+ * |                       |------ Change entry 3
+ * |                       \------ Change entry 4
+ * |
+ * \------- 0.0.1
+ * |--------- Features
+ * |            |------ Change entry 1
+ * |            \------ Change entry 2
+ * |
+ * \--------- Bug fixes
+ * |------ Change entry 3
+ * \------ Change entry 4
```

Comment:
I think this needs some reformatting; `Features` and `Bug fixes` should go under `0.0.1`, change entries 3+4 under `Bug fixes`, right?

I also like the `├──` and `└` below a bit better :)

Reply:
I think the auto formatter broke my ascii art. I’ll see if I can fix it.

Reply:
I fixed the alignment, but will leave the different ascii art. This one is meant to reflect the conceptual structure describing the document model, and the unicode chars are from the `tree` command, representing the concrete directory structure. Thanks again for the careful review!

Reply:
Makes sense, thanks!